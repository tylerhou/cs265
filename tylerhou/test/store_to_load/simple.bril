@main {
    zero: int = const 0;
    four: int = const 4;
    a: ptr<int> = alloc four;
    b: ptr<int> = alloc four;
    c: ptr<int> = ptradd a zero;

    d: ptr<ptr<int>> = alloc four;
    store d a;
    e: ptr<int> = load d;

    free a;
    free b;
    free d;
}
