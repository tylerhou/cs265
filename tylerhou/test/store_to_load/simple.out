@main {
._special_entry_for_ssa:
  zero.0: int = const 0;
  four.0: int = const 4;
  a.0: ptr<int> = alloc four.0;
  b.0: ptr<int> = alloc four.0;
  c.0: ptr<int> = ptradd a.0 zero.0;
  d.0: ptr<ptr<int>> = alloc four.0;
  store d.0 a.0;
  e.0: ptr<int> = id a.0;
  free a.0;
  free b.0;
  free d.0;
}
